sqlx-postgres/src/types/geometry/line_segment.rs

```diff
@@ -0,0 +1,282 @@
+use crate::decode::Decode;
+use crate::encode::{Encode, IsNull};
+use crate::error::BoxDynError;
+use crate::types::Type;
+use crate::{PgArgumentBuffer, PgHasArrayType, PgTypeInfo, PgValueFormat, PgValueRef, Postgres};
+use sqlx_core::bytes::Buf;
+use std::str::FromStr;
+
+const ERROR: &str = "error decoding LSEG";
+
+/// ## Postgres Geometric Line Segment type
+///
+/// Description: Finite line segment
+/// Representation: `((start_x,start_y),(end_x,end_y))`
+///
+///
+/// Line segments are represented by pairs of points that are the endpoints of the segment. Values of type lseg are specified using any of the following syntaxes:
+/// ```text
+/// [ ( start_x , start_y ) , ( end_x , end_y ) ]
+/// ( ( start_x , start_y ) , ( end_x , end_y ) )
+///   ( start_x , start_y ) , ( end_x , end_y )
+///     start_x , start_y   ,   end_x , end_y
+/// ```
+/// where `(start_x,start_y) and (end_x,end_y)` are the end points of the line segment.
+///
+/// See https://www.postgresql.org/docs/16/datatype-geometric.html#DATATYPE-LSEG
+#[derive(Debug, Clone, PartialEq)]
+pub struct PgLSeg {
```

Comment:
The naming is a bit unfortunate since it suggests a relationship to the [`ltree` extension](https://www.postgresql.org/docs/current/ltree.html).

I agree that it's better to be consistent with the convention of matching the name of the Postgres type, but I might suggest adding `#[doc(alias = "line segment")]` or something like that to make it easier to find the type when searching the docs.

Reply:
No strong feelings about naming - even though PgLSeg is more consistent with docs maybe something like `PgLineSegment` would actually be more practical for consumers?
Have included the doc alias, but happy to change.